test/logflare/telemetry_test.exs

```diff
@@ -0,0 +1,94 @@
+defmodule Logflare.TelemetryTest do
+  use Logflare.DataCase
+  alias Logflare.Telemetry
+
+  describe "process metrics" do
+    test "retrieves and emits top 10 by memory" do
+      event = [:logflare, :system, :top_processes, :memory]
+      attach_forwarder(event)
+
+      Telemetry.process_memory_metrics()
+
+      1..10
+      |> Enum.each(fn _ ->
+        assert_receive {:telemetry_event, ^event, metrics, meta}
+
+        assert match?(%{size: _}, metrics)
+        assert match?(%{name: _}, meta)
+      end)
+
+      refute_receive {:telemetry_event, ^event, _, _}
+    end
+
+    test "retrieves and emits top 10 by message queue" do
+      event = [:logflare, :system, :top_processes, :message_queue]
+      attach_forwarder(event)
+
+      Telemetry.process_message_queue_metrics()
+
+      1..10
+      |> Enum.each(fn _ ->
+        assert_receive {:telemetry_event, ^event, metrics, meta}
+
+        assert match?(%{length: _}, metrics)
+        assert match?(%{name: _}, meta)
+      end)
+
+      refute_receive {:telemetry_event, ^event, _, _}
+    end
+  end
+
+  describe "ets_table_metrics/1" do
+    test "retrieves and emits top 10 by table size" do
+      event = [:logflare, :system, :top_ets_tables, :individual]
+      attach_forwarder(event)
+
+      Telemetry.ets_table_metrics()
+
+      1..10
+      |> Enum.each(fn _ ->
+        assert_receive {:telemetry_event, ^event, metrics, meta}
+
+        assert match?(%{size: _}, metrics)
+        assert match?(%{name: _}, meta)
+      end)
+
+      refute_receive {:telemetry_event, ^event, _, _}
```

Comment:
```suggestion
      assert_receive {:telemetry_event, ^event, %{size: _}, %{name: _}}
      refute_receive {:telemetry_event, ^event, _, _}
```
iterating 10 times will not remove it from the inbox so the Enum.each is redundant.
Doing a simple assertion that it is there is sufficient.